src/annotator/test/guest-test.js

```diff
@@ -0,0 +1,869 @@
+    });
+
+    it('marks an annotation in which the target fails to anchor as an orphan', () => {
+      const guest = createGuest();
+      const annotation = {
+        target: [
+          { selector: [{ type: 'TextQuoteSelector', exact: 'notinhere' }] },
+        ],
+      };
+      htmlAnchoring.anchor.returns(Promise.reject());
+
+      return guest
+        .anchor(annotation)
+        .then(() => assert.isTrue(annotation.$orphan));
+    });
+
+    it('marks an annotation in which all (suitable) targets fail to anchor as an orphan', () => {
+      const guest = createGuest();
+      const annotation = {
+        target: [
+          { selector: [{ type: 'TextQuoteSelector', exact: 'notinhere' }] },
+          { selector: [{ type: 'TextQuoteSelector', exact: 'neitherami' }] },
+        ],
+      };
+      htmlAnchoring.anchor.returns(Promise.reject());
+
+      return guest
+        .anchor(annotation)
+        .then(() => assert.isTrue(annotation.$orphan));
+    });
+
+    it('marks an annotation where the target has no TextQuoteSelectors as an orphan', () => {
+      const guest = createGuest();
+      const annotation = {
+        target: [
+          { selector: [{ type: 'TextPositionSelector', start: 0, end: 5 }] },
+        ],
+      };
+      // This shouldn't be called, but if it is, we successfully anchor so that
+      // this test is guaranteed to fail.
+      htmlAnchoring.anchor.returns(Promise.resolve(range));
+
+      return guest
+        .anchor(annotation)
+        .then(() => assert.isTrue(annotation.$orphan));
+    });
+
+    it('does not attempt to anchor targets which have no TextQuoteSelector', () => {
+      const guest = createGuest();
+      const annotation = {
+        target: [
+          { selector: [{ type: 'TextPositionSelector', start: 0, end: 5 }] },
+        ],
+      };
+
+      return guest
+        .anchor(annotation)
+        .then(() => assert.notCalled(htmlAnchoring.anchor));
+    });
+
+    it('updates the cross frame and bucket bar plugins', () => {
+      const guest = createGuest();
+      guest.plugins.CrossFrame = { sync: sinon.stub() };
+      guest.plugins.BucketBar = { update: sinon.stub() };
+      const annotation = {};
+      return guest.anchor(annotation).then(() => {
+        assert.called(guest.plugins.BucketBar.update);
+        assert.called(guest.plugins.CrossFrame.sync);
+      });
+    });
+
+    it('returns a promise of the anchors for the annotation', () => {
+      const guest = createGuest();
+      const highlights = [document.createElement('span')];
+      htmlAnchoring.anchor.returns(Promise.resolve(range));
+      highlighter.highlightRange.returns(highlights);
+      const target = {
+        selector: [{ type: 'TextQuoteSelector', exact: 'hello' }],
+      };
+      return guest
+        .anchor({ target: [target] })
+        .then(anchors => assert.equal(anchors.length, 1));
+    });
+
+    it('adds the anchor to the "anchors" instance property"', () => {
```

Comment:
This feels a little implementation-detail-y...I'm curious what this is attempting to test at a higher level.

Reply:
The `anchors` property is currently part of the public interface of `Guest` which `bucket-bar.coffee` and possibly others rely on. The code design could be improved but this test is currently appropriately testing part of the public interface.